### ISSUE TITLE
`Series.mul(-1)` changes Series column types from `float32` to `int32` on Series with leading zeros

### ISSUE THREAD

**Issue by licarth** [opensource9ja/danfojs#251]
**Describe the bug**
`Series.mul(-1)` rounds floats to the nearest integer when run on a Series with leading zeros (more than 10).
More generally, Series with more than 10 leading zeros are incorrectly typed (infered column types are wrong).
More annoyingly, even when setting col_types manually, `.mul()` switches back to `int32`.

See tests below.

**To Reproduce**

```typescript
import { Series } from "danfojs-node";
import { toBeDeepCloseTo } from "jest-matcher-deep-close-to";
expect.extend({ toBeDeepCloseTo });
declare global {
  namespace jest {
    interface Matchers<R> {
      toBeDeepCloseTo(value: number[], decimals?: number): CustomMatcherResult;
    }
  }
}
describe("Series.mul()", () => {
  describe("should not round to integers when multiplying by -1", () => {
    it("of a short Series", () => {
      const originalSeries = new Series([100.1]);
      const expectedOpposite = new Series([-100.1]);
      const seriesOpposite = originalSeries.mul(-1);
      expect(seriesOpposite.values).toBeDeepCloseTo(expectedOpposite.values, 4); // ✅️ PASS
    });

    it("of a Series with 10 or more leading zeros", () => {
      const originalSeries = new Series([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 100.1]);
      const expectedOpposite = new Series([
        -0, -0, -0, -0, -0, -0, -0, -0, -0, -0, -100.1,
      ]);
      const seriesOpposite = originalSeries.mul(-1);
      expect(seriesOpposite.values).toBeDeepCloseTo(expectedOpposite.values, 4); // ❌️ FAIL
      // expect(received).toBeDeepCloseTo(expected)
      //
      // Expected value to be (using 4 decimals):
      //   -100.1
      // Received:
      //   -100
    });
  });

  describe("should correctly infer column types", () => {
    it("of a short Series", () => {
      const series = new Series([100.1]);
      expect(series.col_types).toEqual(["float32"]); // ✅️ PASS
    });

    it("of a Series with 10 or more leading zeros", () => {
      const series = new Series([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 100.1]);
      expect(series.col_types).toEqual(["float32"]); // ❌️ FAIL
      // expect(received).toEqual(expected) // deep equality
      //
      // - Expected  - 1
      // + Received  + 1
      //   Array [
      // -   "float32",
      // +   "int32",
      //   ]
    });
  });

  describe("should not change the col_types if manually defined", () => {
    it("of a short Series", () => {
      const originalSeries = new Series([100.1]);
      originalSeries._set_col_types(["float32"], false);
      const seriesOpposite = originalSeries.mul(-1);
      expect(originalSeries.col_types).toEqual(["float32"]); // ✅️ PASS
      expect(seriesOpposite.col_types).toEqual(["float32"]); // ✅️ PASS
    });

    it("of a Series with 10 or more leading zeros", () => {
      const originalSeries = new Series([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 100.1]);
      originalSeries._set_col_types(["float32"], false);
      const seriesOpposite = originalSeries.mul(-1);
      expect(originalSeries.col_types).toEqual(["float32"]); // ✅️ PASS
      expect(seriesOpposite.col_types).toEqual(["float32"]); // ❌️ FAIL
      // expect(received).toEqual(expected) // deep equality
      //
      // - Expected  - 1
      // + Received  + 1
      //   Array [
      // -   "float32",
      // +   "int32",
      //   ]
    });
  });
});
```

**Expected behavior**
- Type inference should work with leading zeros
- `mul()` should not change column type from `float32` to `int32`.

**Comment by risenW** [comment]
@licarth Thanks for taking out time to report this, and the test case you provided will really help. This is definitely a bug, and I'll look into it ASAP.

**Comment by risenW** [comment]
@licarth This bug on `mul` function has now been fixed in the latest version (v0.3.3)

Note: You can set type on creation by doing the following:
```javascript
const series = new dfd.Series([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 100.1], { dtypes: ["float32"] });
console.log(series.dtypes);
```
